src/mysql/connection.cr

```diff
@@ -1,6 +1,37 @@
 require "socket"
 
 class MySql::Connection < DB::Connection
+  class ConnectionError < Exception; end
+
+  class PacketError < ConnectionError
+    getter :packet
+
+    def initialize(packet : ReadPacket, message)
```

Comment:
I don't think ReadPacket should be exposed, it's an internal implementation detail.

Reply:
Check out the changes on c14f4e6532e5005307e57f0458212297146db3b6; I'm hoping it addresses your concern.